src/cmdstan-guide/pathfinder_intro.Rmd

```diff
@@ -0,0 +1,129 @@
+for everything but the location of the data file.
+
+```
+> ./bernoulli pathfinder data file=bernoulli.data.R
+```
+
+Executing this command prints both output to the console and csv files.
+
+The first part of the console output reports on the configuration used.
+```
+method = pathfinder
+  pathfinder
+    init_alpha = 0.001 (Default)
+    tol_obj = 9.9999999999999998e-13 (Default)
+    tol_rel_obj = 10000 (Default)
+    tol_grad = 1e-08 (Default)
+    tol_rel_grad = 10000000 (Default)
+    tol_param = 1e-08 (Default)
+    history_size = 5 (Default)
+    num_psis_draws = 1000 (Default)
+    num_paths = 4 (Default)
+    max_lbfgs_iters = 1000 (Default)
+    num_draws = 1000 (Default)
+    num_elbo_draws = 25 (Default)
+id = 1 (Default)
+data
+  file = examples/bernoulli/bernoulli.data.json
+init = 2 (Default)
+random
+  seed = 1995513073 (Default)
+output
+  file = output.csv (Default)
+  diagnostic_file =  (Default)
+  refresh = 100 (Default)
+  sig_figs = -1 (Default)
+  profile_file = profile.csv (Default)
+num_threads = 1 (Default)
+```
+The rest of the output describes the progression of the algorithm.
+
+By default, the Pathfinder algorithm runs 4 single-path Pathfinders in parallel, the uses importance resampling on the set of returned draws to produce the specified number of draws.
+```
+Path [1] :Initial log joint density = -11.543343
+Path [1] : Iter      log prob        ||dx||      ||grad||     alpha      alpha0      # evals       ELBO    Best ELBO        Notes 
+              5      -6.748e+00      1.070e-03   1.707e-05    1.000e+00  1.000e+00       126 -6.220e+00 -6.220e+00                  
+Path [1] :Best Iter: [5] ELBO (-6.219833) evaluations: (126)
+Path [2] :Initial log joint density = -7.443345
+Path [2] : Iter      log prob        ||dx||      ||grad||     alpha      alpha0      # evals       ELBO    Best ELBO        Notes 
+              5      -6.748e+00      9.936e-05   3.738e-07    1.000e+00  1.000e+00       126 -6.164e+00 -6.164e+00                  
+Path [2] :Best Iter: [5] ELBO (-6.164015) evaluations: (126)
+Path [3] :Initial log joint density = -18.986308
+Path [3] : Iter      log prob        ||dx||      ||grad||     alpha      alpha0      # evals       ELBO    Best ELBO        Notes 
+              5      -6.748e+00      2.996e-04   4.018e-06    1.000e+00  1.000e+00       126 -6.201e+00 -6.201e+00                  
+Path [3] :Best Iter: [5] ELBO (-6.200559) evaluations: (126)
+Path [4] :Initial log joint density = -8.304453
+Path [4] : Iter      log prob        ||dx||      ||grad||     alpha      alpha0      # evals       ELBO    Best ELBO        Notes 
+              5      -6.748e+00      2.814e-04   2.034e-06    1.000e+00  1.000e+00       126 -6.221e+00 -6.221e+00                  
+Path [4] :Best Iter: [3] ELBO (-6.161276) evaluations: (126)
+Total log probability function evaluations:8404
+```
+
+Pathfinder output is written into a set of StanCSV files; one file which contains the importance resampled draws
```

Comment:
Whare are "StanCSV files"?  Can't we just say "CSV" files. 

"one file which" -> "one file that" [which is for non-restrictive relatives!]

"one file which contains" -> "one file per single-path Pathfinder run" (as written it feels like one file)

Reply:
the CSV format is underspecified.  these files contain comments before the header row and after the data rows; not all CSV parsers will be happy. 
linking to Pathfinder CSV section